boa_engine/src/builtins/intl/tests.rs

```diff
@@ -154,6 +163,13 @@ fn uni_ext_comp() {
     assert_eq!(components.keywords.len(), 1);
     assert_eq!(components.keywords[0].key, "ca");
     assert_eq!(components.keywords[0].value, "islamic-civil");
+
+    let ext = JsString::new("u-ca-islamic-civil");
+    let components = unicode_extension_components(&ext);
```

Comment:
We give it to you - you can just use `locale.extensions.unicode` for all of that. No need to reconstruct it with your own types. https://docs.rs/icu/latest/icu/locid/extensions/unicode/index.html

Reply:
Done. https://github.com/boa-dev/boa/pull/2072/files#diff-1dd4af22d3a6142a8060d056e9d0489d84d8a4d9c68a8f6e7b359b8d3ab35499R425